IBM/carbon-addons-iot-react
# 1220 default value

Closes #1220

**Summary**

Proptype declaration for `defaultValue` proposed a shape that was never used in the component code or in the storybook. This PR changes the prop type declaration to resemble how its being used in the code.

**Change List (commits, features, bugs, etc)**

- DateTimePicker: changed prop type declaration
- DateTimePicker.test.jsx added test cases to see if component would blow up if random default value was passed and passes if correct shape.
- PublicAPIsnapshot: updated snapshots from code change. 

**Acceptance Test (how to verify the PR)**

- Go to DateTimePicker story
- see that the stories for all three value types work and render the right value
- test checking for blow up on wrong value passes.
